lib/new_relic/agent/instrumentation/sidekiq/extensions/delayed_class.rb

```diff
@@ -2,21 +2,23 @@
 # See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
 # frozen_string_literal: true
 
-class Sidekiq::Extensions::DelayedClass
-  def newrelic_trace_args(msg, queue)
-    (target, method_name, _args) = if YAML.respond_to?(:unsafe_load)
-      YAML.unsafe_load(msg['args'][0])
-    else
-      YAML.load(msg['args'][0])
-    end
+if Sidekiq::VERSION < '7.0.0'
```

Comment:
Would you mind adding a TODO here about removing this class when we no longer support Sidekiq 6?

Reply:
Good idea! Submitted https://github.com/newrelic/newrelic-ruby-agent/pull/1616